Sources/TypeChecker/TypeChecker+Expressions.swift

```diff
@@ -130,6 +132,42 @@ extension TypeChecker {
                                       expression: argument.expression))
         }
       }
+    } else if case .matchedFunction(let matchingFunction) =
+      environment.matchFunctionCall(functionCall,
+                                    enclosingType: functionCall.identifier.enclosingType ?? enclosingType,
+                                    typeStates: typeStates,
+                                    callerProtections: callerProtections,
+                                    scopeContext: passContext.scopeContext!) {
+
+      if let externalCall = passContext.externalCall {
+        // check value parameter (type)
+        if matchingFunction.declaration.isPayable {
+          if let valueParameter: FunctionArgument = externalCall.getHyperParameter(parameterName: "value") {
```

Comment:
Perhaps we should make these enums? They are constants that could change, also there's a lot of repetition here